### PR TITLE
fix: KCP registration process may fail when switching workspace

### PR DESCRIPTION
**Pull request by Roming22**
The ServiceAccount used may not have access to the root workspace. Switching
workspaces is unnecessary if we're already in the target workspace.

Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

### CODE REVIEW

**Comment by fgiloux** [comment]
If I am understanding you correctly you are saying that the credentials you are using may not give you access to the organization workspace as you are trying to create the workspace directly under root.
Assuming that in such a case the workspace would be already existing and why `if [[ "$(KUBECONFIG=${kcp_kcfg} kubectl kcp workspace current | cut -d\" -f2)" == "$KCP_ORG:$KCP_WORKSPACE" ]]; then`may work. I think it would make sense not to require to be already in the correct workspace for the code to work for your scenario.
I did a bit of testing with kcp 0.5 and latest and unfortunately it is possible to switch to a workspace that  does not exist (and to create resources there), which is IMHO a flaw in kcp. I will check whether it is tracked somewhere.

TL;DR: I am fine with your change for now but we may need to follow up later on to make it more robust

**Comment by Roming22** [comment]
@fgiloux What I'm saying is that if a user creates the workspace `root:org1:workspace1` and creates a service account in that workspace, the serviceaccount won't have the right to list the workspaces in `root:org1`, so will try to create the workspace `root:org1:workspace1`, which it does not have the right to do.

If you want the code to be a bit more robust, I could change the the if statement to:
```if ! KUBECONFIG="$kcp_kcfg" kubectl kcp workspace use "$KCP_ORG:$KCP_WORKSPACE"; then```

Just let me know.

**Comment by Roming22** [comment]
@fgiloux I agree with your proposal.
